napari/_qt/containers/_layer_delegate.py

```diff
@@ -0,0 +1,156 @@
+"""
+General rendering flow:
+
+1. The List/Tree view needs to display or edit an index in the model...
+2. It gets the ``itemDelegate``
+    a. A custom delegate can be set with ``setItemDelegate``
+    b. ``QStyledItemDelegate`` is the default delegate for all Qt item views,
+       and is installed upon them when they are created.
+3. ``itemDelegate.paint`` is called on the index being displayed
+4. Each index in the model has various data elements (i.e. name, image, etc..),
+   each of which has a "data role".  A model should return the appropriate data
+   for each role by reimplementing ``QAbstractItemModel.data``.
+    a. `QStyledItemDelegate` implements display and editing for the most common
+       datatypes expected by users, including booleans, integers, and strings.
+    b. If the delegate does not support painting of the data types you need or
+       you want to customize the drawing of items, you need to subclass
+       ``QStyledItemDelegate``, and reimplement ``paint()`` and possibly
+       ``sizeHint()``.
+    c. When reimplementing ``paint()``, one typically handles the datatypes
+       they would like to draw and uses the superclass implementation for other
+       types.
+5. The default implementation of ``QStyledItemDelegate.paint`` paints the item
+   using the view's ``QStyle`` (which is, by default, an OS specific style...
+   but see ``QCommonStyle`` for a generic implementation)
+    a. It is also possible to override the view's style, using either a
+    subclass of ``QCommonStyle``, for a platform-independent look and feel, or
+    ``QProxyStyle``, which let's you override only certain stylistic elements
+    on any platform, falling back to the system default otherwise.
+    b. ``QStyle`` paints various elements using methods like ``drawPrimitive``
+       and ``drawControl``.  These can be overridden for very fine control.
+6. It is hard to use stylesheets with custom ``QStyles``... but it's possible
+   to style sub-controls in ``QAbstractItemView`` (such as ``QTreeView``):
+   https://doc.qt.io/qt-5/stylesheet-reference.html#list-of-sub-controls
+
+"""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from qtpy import QtCore
+from qtpy.QtCore import QSize, Qt
+from qtpy.QtGui import QPixmap
+from qtpy.QtWidgets import QStyledItemDelegate
+
+from ..qt_resources import QColoredSVGIcon
+
+if TYPE_CHECKING:
+    from qtpy.QtCore import QModelIndex
+    from qtpy.QtGui import QPainter
+    from qtpy.QtWidgets import QStyleOptionViewItem, QWidget
+
+
+class LayerDelegate(QStyledItemDelegate):
```

Comment:
Not sure if done here or in stylesheet, but can we make the height of each of the items a little less, they seem a bit taller to me than before, but in general having them thinner will make more visible at any moment. Happy to have that in a followup PR, but noting now